File: composeApp/src/commonMain/kotlin/com/daddoodev/yetimatch/ui/screens/QuizScreen.kt
package com.daddoodev.yetimatch.ui.screens

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import com.daddoodev.yetimatch.models.Answer
import com.daddoodev.yetimatch.models.Question
import com.daddoodev.yetimatch.viewmodels.QuizViewModel

@Composable
fun QuizScreen(
    viewModel: QuizViewModel,
    onQuizComplete: () -> Unit = {}
) {
    val currentQuiz by viewModel.currentQuiz.collectAsState()
    val currentQuestionIndex by viewModel.currentQuestionIndex.collectAsState()
    val isQuizComplete by viewModel.isQuizComplete.collectAsState()
    val progress = viewModel.getProgress()
    
    LaunchedEffect(isQuizComplete) {
        if (isQuizComplete) {
            onQuizComplete()
        }
    }
    
    currentQuiz?.let { quiz ->
        val currentQuestion = quiz.questions.getOrNull(currentQuestionIndex)
        
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(16.dp)
        ) {
            // Progress bar
            LinearProgressIndicator(
                progress = { progress / 100f },
                modifier = Modifier.fillMaxWidth(),
            )
            
            Spacer(modifier = Modifier.height(16.dp))
            
            // Question counter
            Text(
                text = "Question ${currentQuestionIndex + 1} of ${quiz.questions.size}",
                style = MaterialTheme.typography.labelLarge,
                color = MaterialTheme.colorScheme.primary
            )
            
            Spacer(modifier = Modifier.height(24.dp))
            
            // Question text
            currentQuestion?.let { question ->
                Text(
                    text = question.text,
                    style = MaterialTheme.typography.headlineSmall,
                    textAlign = TextAlign.Start
                )
                
                Spacer(modifier = Modifier.height(32.dp))
                
                // Answer options
                LazyColumn(
                    verticalArrangement = Arrangement.spacedBy(12.dp),
                    modifier = Modifier.weight(1f)
                ) {
                    items(
                        items = question.answers,
                        key = { it.id }
                    ) { answer ->
                        AnswerCard(
                            answer = answer,
                            onAnswerSelected = { viewModel.answerQuestion(answer.id) }
                        )
                    }
                }
                
                Spacer(modifier = Modifier.height(16.dp))
                
                // Back button (if not first question)
                if (currentQuestionIndex > 0) {
                    OutlinedButton(
                        onClick = { viewModel.previousQuestion() },
                        modifier = Modifier.fillMaxWidth()
                    ) {
                        Text("Previous Question")
                    }
                }
            }
        }
    }
}

@Composable
fun AnswerCard(
    answer: Answer,
    onAnswerSelected: () -> Unit
) {
    Card(
        onClick = onAnswerSelected,
        modifier = Modifier.fillMaxWidth()
    ) {
        Text(
            text = answer.text,
            style = MaterialTheme.typography.bodyLarge,
            modifier = Modifier.padding(16.dp)
        )
    }
}

File: lib/screens/quiz_screen.dart
import 'package:flutter/material.dart';

import '../state/quiz_state.dart';

class QuizScreen extends StatelessWidget {
  const QuizScreen({
    super.key,
    required this.quizState,
    required this.onQuizComplete,
  });

  final QuizState quizState;
  final VoidCallback onQuizComplete;

  @override
  Widget build(BuildContext context) {
    return ListenableBuilder(
      listenable: quizState,
      builder: (context, _) {
        if (quizState.isQuizComplete) {
          WidgetsBinding.instance.addPostFrameCallback((_) => onQuizComplete());
        }
        final quiz = quizState.currentQuiz;
        if (quiz == null) return const SizedBox.shrink();
        final questionIndex = quizState.currentQuestionIndex;
        final question = questionIndex < quiz.questions.length
            ? quiz.questions[questionIndex]
            : null;
        if (question == null) return const SizedBox.shrink();

        final progress = quizState.getProgress() / 100;

        return Padding(
          padding: const EdgeInsets.all(16),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.stretch,
            children: [
              LinearProgressIndicator(value: progress),
              const SizedBox(height: 16),
              Text(
                'Question ${questionIndex + 1} of ${quiz.questions.length}',
                style: Theme.of(context).textTheme.labelLarge?.copyWith(
                      color: Theme.of(context).colorScheme.primary,
                    ),
              ),
              const SizedBox(height: 24),
              Text(
                question.text,
                style: Theme.of(context).textTheme.headlineSmall,
              ),
              const SizedBox(height: 32),
              Expanded(
                child: ListView.separated(
                  itemCount: question.answers.length,
                  separatorBuilder: (_, i) => const SizedBox(height: 12),
                  itemBuilder: (_, i) {
                    final answer = question.answers[i];
                    return Card(
                      child: ListTile(
                        title: Text(answer.text),
                        onTap: () => quizState.answerQuestion(answer.id),
                      ),
                    );
                  },
                ),
              ),
              if (questionIndex > 0) ...[
                const SizedBox(height: 16),
                OutlinedButton(
                  onPressed: () => quizState.previousQuestion(),
                  child: const Text('Previous Question'),
                ),
              ],
            ],
          ),
        );
      },
    );
  }
}
